core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -253,7 +261,8 @@ public void testSerialization() throws IOException {
         try (BytesStreamOutput output = new BytesStreamOutput()) {
             testQuery.writeTo(output);
             try (StreamInput in = StreamInput.wrap(output.bytes())) {
-                QB deserializedQuery = createEmptyQueryBuilder().readFrom(in);
+                QueryBuilder prototype = queryParserService.queryParser(testQuery.queryId()).getBuilderPrototype();
+                QueryBuilder deserializedQuery = prototype.readFrom(in);
```

Comment:
When changing this I also realized we need the parser for getPrototype() at the moment. At some point it might make sense to introduce similar getter to the builder though. Do you think we should keep that in mind as an option? Didn't do it here because I don't want to touch every builder again with this PR.